### PR TITLE
KAFKA-7002: Add a config property for DLQ topic's replication factor (KIP-298)

### PR DESCRIPTION
**Pull request by wicknicks**
Currently, the replication factor is hardcoded to a value of 3. This means that we cannot use a DLQ in any cluster setup with less than three brokers. It is better to have the user specify this value if the default value does meet the requirements.

Testing: A unit test is added.

Signed-off-by: Arjun Satish <arjun@confluent.io>

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by wicknicks** [comment]
@mageshn @rayokota @rhauch any thoughts on this PR?

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SinkConnectorConfig.java [hunk 1]
```diff
@@ -52,10 +52,16 @@
     public static final String DLQ_TOPIC_DEFAULT = "";
     private static final String DLQ_TOPIC_DISPLAY = "Dead Letter Queue Topic Name";
 
+    public static final String DLQ_TOPIC_REPLICATION_FACTOR_CONFIG = DLQ_PREFIX + "topic.replication.factor";
+    private static final String DLQ_TOPIC_REPLICATION_FACTOR_CONFIG_DOC = "Replication factor used when creating the dead letter queue topic";
```
Comment (rhauch) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SinkConnectorConfig.java]: Might want to add that this is not used if the topic already exists.

**Review by rhauch — APPROVED** [review]
One very minor but optional suggestion. Otherwise LGTM. Thanks!

**Comment by wicknicks** [comment]
@ewencp could you please take a look at this PR, and merge it if things look good? Thanks!

**Review by ewencp — APPROVED** [review]
This looks fine, but in our example configs we also set other replication factors to values that will work for a single, local instance even in distributed mode. Should we include that as well? It's more critical for the ones that are in that file because they are required to even start up, but seems like if we're sticking to example values that work on a single node "cluster", we'd want to make sure all features would work there.

**Comment by wicknicks** [comment]
@ewencp happy to update the example. But as of now, DLQ is disabled by default. if we add this config in the example files, we will need to enable error handling, enable DLQ and then set the replication factor. I think it is ok to exclude it from the current example for the single node setup. Thoughts?

**Comment by ewencp** [comment]
Hmm, fair point, guess it kinda sucks to have to set the parent one as well. Let me merge this for now, and we can follow up on whether having this default but not overriding it in the example configs might cause problems.

Only semi-related, but we might also want to think about adding some production-grade example configs at some point.

Merging as is to trunk and 2.0, and we can follow up separately if we need to.

**Pull request closed without merging**